utilities.php

```diff
@@ -64,12 +64,61 @@ function print_listing_li($item_id, $title, $currentPrice, $num_bids, $end_time)
   );
 }
 
+// print_mylisting_li:
+// This function prints an HTML <li> element containing an auction listing
+function print_my_listings_li($item_id, $title, $currentPrice, $num_bids, $end_time, $auction_status)
+{
+
+  // Fix language of bid vs. bids
+  if ($num_bids == 1) {
+    $bid = ' bid';
+  } else {
+    $bid = ' bids';
+  }
+
+  // Calculate time to auction end
+  $now = new DateTime();
+  if ($now > $end_time) {
+    $time_remaining = 'This auction has ended';
+  } else {
+    // Get interval:
+    $time_to_end = date_diff($now, $end_time);
+    $time_remaining = display_time_remaining($time_to_end) . ' remaining';
+  }
+
+  // Conditional colouring for auction status
+  $status_colour = match($auction_status) {

```

Comment:
think I found a bug here - I created an auction and when the time passed and there was no bids. The website shows fatal error like the following. No default was set here
<img width="1123" alt="Screenshot 2024-11-15 at 16 18 13" src="https://github.com/user-attachments/assets/d9cc7ee9-0464-4f22-8982-20d7f2968c7c">

Reply:
Resolved, updated utilities function and sql query